nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -144,28 +164,23 @@ class NodeletLazy: public nodelet::Nodelet
     if (lazy_)
     {
       boost::mutex::scoped_lock lock(connection_mutex_);
-      for (size_t i = 0; i < publishers_.size(); i++)
+      if (hasSubscribers())
       {
-        ros::Publisher pub = publishers_[i];
-        if (pub.getNumSubscribers() > 0)
+        if (connection_status_ != SUBSCRIBED)
         {
-          if (connection_status_ != SUBSCRIBED)
+          if (verbose_connection_)
           {
-            if (verbose_connection_)
-            {
-              NODELET_INFO("Subscribe input topics");
-            }
-            subscribe();
-            connection_status_ = SUBSCRIBED;
+            NODELET_INFO("Subscribe input topics");
           }
-          if (!ever_subscribed_)
-          {
-            ever_subscribed_ = true;
-          }
-          return;
+          subscribe();
+          connection_status_ = SUBSCRIBED;
+        }
+        if (!ever_subscribed_)
+        {
+          ever_subscribed_ = true;
         }
       }
-      if (connection_status_ == SUBSCRIBED)
+      else if (connection_status_ == SUBSCRIBED)
```

Comment:
The rest remains the same, but I changed the `return;` inside the `if` block to and `if-else` block, which is equivalent and easier to read in my opinion. If you don't like it, I can remove this from the PR.